lib/tool_shed/webapp/frontend/src/components/RegisterPage.vue

```diff
@@ -12,19 +12,23 @@ const confirm = ref("")
 const username = ref("")
 
 const title = ref("Register")
-const createFetcher = fetcher.path("/api_internal/register").method("post").create()
 // type Response = components["schemas"]["UiRegisterResponse"]
 
 async function onRegister() {
     // TODO: handle confirm and implement bear_field.
     // let data: Response
     try {
-        const { data } = await createFetcher({
-            email: email.value,
-            password: password.value,
-            username: username.value,
-            bear_field: "",
+        const { data } = await client.POST("/api_internal/register", {
+            body: {
+                email: email.value,
+                password: password.value,
+                username: username.value,
+                bear_field: "",
+            },
         })
+        if (!data) {
+            throw new Error("No data returned")
```

Comment:
That's similar to https://github.com/galaxyproject/galaxy/pull/18532#discussion_r1713624420

Reply:
I think this line will never be reached because of the middleware. It could be just replaced with an empty return or a check for data is defined before using it.